src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,504 @@
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public Tuple<long, long> DataGetTime(int data)
+        {
+            long startTime, endTime;
+
+            int error = Interop.SensorRecoder.DataGetTime(data, out startTime, out endTime);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error getting the sensor recorder time");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.DataGetTime Failed");
+            }
+
+            Tuple<long, long> list = new Tuple<long, long>(startTime, endTime);
+            return list;
+        }
+
+        /// <summary>
+        /// Checks whether it is supported to record a given sensor type.
+        /// </summary>
+        /// <param name="type">
+        /// A sensor type to check.
+        /// </param>
+        /// <returns>
+        /// returns true if it is supported to record a given sensor type.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public bool IsSupported(int type)
+        {
+            bool isSupported;
+            int error = Interop.SensorRecoder.IsSupported(type, out isSupported);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error checking if sensor recorder is supported");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.IsSupported Failed");
+            }
+            return isSupported;
+        }
+
+        /// <summary>
+        /// Sets a recording option parameter.
+        /// </summary>
+        /// <param name="recorderOption">
+        /// Option parameter.
+        /// </param>
+        /// <param name="value">
+        /// Value.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void OptionSetInt(int recorderOption, int value)
+        {
+            if (CheckOption())
+            {
+                int error = Interop.SensorRecoder.OptionSetInt(OptionHandle, recorderOption, value);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error checking sensor recorder option set int");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.OptionSetInt Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder option is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Sets an integer-type query parameter.
+        /// </summary>
+        /// <param name="recorderQuery">
+        /// Query parameter.
+        /// </param>
+        /// <param name="value">
+        /// Value.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void QuerySetInt(int recorderQuery, int value)
```

Comment:
What is recorderQuery and why is it a int type?

Reply:
its an Enum for filtering and aggregation parameters for querying sensor records. I will replace it with enum.